src/server/hset_family.cc

```diff
@@ -720,19 +721,44 @@ OpResult<vector<long>> OpHExpire(const OpArgs& op_args, string_view key, uint32_
   return HSetFamily::SetFieldsExpireTime(op_args, ttl_sec, key, values, pv);
 }
 
-// HSETEX key [NX] tll_sec field value field value ...
+enum class HSetExOpts {
+  NX,
+  KEEPTTL,
+};
+
+OpSetParams LoadHSetExParams(CmdArgParser& parser) {
+  OpSetParams op_sp{};
+
+  auto res = parser.TryMapNext("NX"sv, HSetExOpts::NX, "KEEPTTL"sv, HSetExOpts::KEEPTTL);
+  while (res.has_value()) {
+    switch (res.value()) {
+      case HSetExOpts::NX:
+        op_sp.skip_if_exists = true;
+        break;
+      case HSetExOpts::KEEPTTL:
+        op_sp.keepttl = true;
+        break;
+    }
+    res = parser.TryMapNext("NX"sv, HSetExOpts::NX, "KEEPTTL"sv, HSetExOpts::KEEPTTL);
+  }
+
+  // Conversion errors are handled by caller
+  op_sp.ttl = parser.Next<uint32_t>();
+  return op_sp;
+}
```

Comment:
Do we really need all of this ? Why not `bool skip_if_exists = static_cast<bool>(parser.Check("KEEPTTL"sv));` similarly to what we have on line 729 ? (see `skip_if_exists`)

Reply:
I originally tried with `Check` but it seems to only work for a fixed order, eg if we have:

```
bool keepttl = parser.Check("KEEPTTL"sv);
bool skip = parser.Check("NX"sv);
```

This works if
* only `KEEPTTL` is passed
* only `NX` is passed
* both are passed in the exact order `KEEPTTL NX`

But it rejects the command if we call `HSETEX key NX KEEPTTL ...`, so we have the order hardcoded.

If we reverse the order of `parser.Check` calls then `NX KEEPTTL` will work but `KEEPTTL NX` will not work.

So I used a loop so the current version of the code handles the options in any order.

Reply:
If the order of options is guaranteed to be fixed eg `NX` is always before `KEEPTTL` then I can add the two `parser.Check` calls in the same order.

Reply:
switched to use `Check` in a loop as discussed